From f4aa64aeaf1bdcca8e1c43e763f4b7922bd9bfa3 Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Tue, 16 Sep 2025 14:28:58 +0200
Subject: [PATCH 921/954] fix(stm32mp25-fdts): new swizzle configuration for
 STM32MP257F-EV1 board

Correction of the bank group management for x8 devices to remove
impacts on other cases.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 718c1495c604034d2122c5940ad20fcb74e1018e fix(stm32mp25-fdts): new swizzle configuration for STM32MP257F-EV1 board]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: Iec46a35bd5e83f8de88a13fce29357b2b10c2a77
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/490324
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 fdts/stm32mp25-ddr4-2x16Gbits-2x16bits-1200MHz.dtsi | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/fdts/stm32mp25-ddr4-2x16Gbits-2x16bits-1200MHz.dtsi b/fdts/stm32mp25-ddr4-2x16Gbits-2x16bits-1200MHz.dtsi
index d0a6be7b56..b56891aa8c 100644
--- a/fdts/stm32mp25-ddr4-2x16Gbits-2x16bits-1200MHz.dtsi
+++ b/fdts/stm32mp25-ddr4-2x16Gbits-2x16bits-1200MHz.dtsi
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause
 /*
- * Copyright (C) 2022-2024, STMicroelectronics - All Rights Reserved
+ * Copyright (C) 2022-2025, STMicroelectronics - All Rights Reserved
  */
 
 /*
@@ -223,7 +223,7 @@
 #define DDR_UIS_SWIZZLE_19 0x0000000B
 #define DDR_UIS_SWIZZLE_20 0x00000000
 #define DDR_UIS_SWIZZLE_21 0x00000000
-#define DDR_UIS_SWIZZLE_22 0x00000000
+#define DDR_UIS_SWIZZLE_22 0x00000011
 #define DDR_UIS_SWIZZLE_23 0x00000008
 #define DDR_UIS_SWIZZLE_24 0x00000002
 #define DDR_UIS_SWIZZLE_25 0x00000018
@@ -232,7 +232,7 @@
 #define DDR_UIS_SWIZZLE_28 0x0D0A0407
 #define DDR_UIS_SWIZZLE_29 0x00000014
 #define DDR_UIS_SWIZZLE_30 0x000B0600
-#define DDR_UIS_SWIZZLE_31 0x02080000
+#define DDR_UIS_SWIZZLE_31 0x02081100
 #define DDR_UIS_SWIZZLE_32 0x00000018
 #define DDR_UIS_SWIZZLE_33 0x00000000
 #define DDR_UIS_SWIZZLE_34 0x00000000
-- 
2.53.0

